app/models/category.rb

```diff
@@ -0,0 +1,11 @@
+class Category < ApplicationRecord
+  validates :name, presence: true
+  validates :icon, presence: true, length: { maximum: 2 }
+
+  belongs_to :user
+  has_and_belongs_to_many :activities
+
+  def total_amount
+    activities.sum(:amount)
+  end
+end
```

Comment:
- [ ] Please validate the input fields for the `Category and transaction`. You have to flash or raise the message if a user tries to create the data without the information or with the empty data.